core/src/usb.py

```diff
@@ -1,19 +1,46 @@
 from storage.device import get_device_id
 from trezor import io, utils
 
+_iface_iter = iter(range(5))
+
+ENABLE_IFACE_DEBUG = __debug__
+# change to False to enable VCP, see below
+ENABLE_IFACE_WEBAUTHN = not utils.BITCOIN_ONLY
+
+# We only have 10 available USB endpoints on real HW, of which 2 are taken up by the USB descriptor.
+# iface_wire, iface_debug and iface_webauthn also consume 2 each, iface_vcp uses 3.
+# That is a grand total of 11. That means that we can't enable everything at the same time.
+# By default, iface_vcp is only enabled on bitcoin_only firmware, where iface_webauthn
+# is disabled. Implementation-wise, we check if any of the previous ifaces is disabled
+# in order to enable VCP.
+ENABLE_IFACE_VCP = __debug__ and (
+    utils.EMULATOR or not ENABLE_IFACE_DEBUG or not ENABLE_IFACE_WEBAUTHN
+)
+
 # interface used for trezor wire protocol
-iface_wire = io.WebUSB(iface_num=0, ep_in=0x81, ep_out=0x01)
+id_wire = next(_iface_iter)
+iface_wire = io.WebUSB(
+    iface_num=id_wire,
+    ep_in=0x81 + id_wire,
+    ep_out=0x01 + id_wire,
+)
 
-if __debug__:
+if __debug__ and ENABLE_IFACE_DEBUG:
     # interface used for debug messages with trezor wire protocol
-    iface_debug = io.WebUSB(iface_num=1, ep_in=0x82, ep_out=0x02)
+    id_debug = next(_iface_iter)
+    iface_debug = io.WebUSB(
+        iface_num=id_debug,
+        ep_in=0x81 + id_debug,
+        ep_out=0x01 + id_debug,
+    )
 
-if not utils.BITCOIN_ONLY:
+if ENABLE_IFACE_WEBAUTHN:
```

Comment:
This is not the same as `not utils.BITCOIN_ONLY`, the compiler won't remove it even when you try to declare it with `const()`.

`if not utils.BITCOIN_ONLY and ENABLE_IFACE_WEBAUTHN` works too, although it looks weird.

Reply:
AFAIK the compiler is not trying to be particularly smart, so this makes sense